scripts/pegasus_online_node_list.sh

```diff
@@ -0,0 +1,118 @@
+cd $shell_dir
+
+echo "Check the argument..."
+source ./scripts/pegasus_check_arguments.sh online_node_list $cluster $meta_list $replica_task_id_list
+
+if [ $? -ne 0 ]; then
+    echo "ERROR: the argument check failed"
+    exit 1
+fi
+
+echo "Set meta level to steady..."
+echo "set_meta_level steady" | ./run.sh shell --cluster $meta_list &>/tmp/$UID.$PID.pegasus.online_node.set_meta_level
+set_ok=`grep 'control meta level ok' /tmp/$UID.$PID.pegasus.online_node.set_meta_level | wc -l`
+if [ $set_ok -ne 1 ]; then
+  echo "ERROR: set meta level to steady failed"
+  exit 1
+fi
+
+for id in $task_id_list
+do
+  echo "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
+  minos_bootstrap $cluster replica $id
+  if [ $? -ne 0 ]; then
+    echo "ERROR: online replica task $id failed"
+    exit 1
+  fi
+  echo "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
+  echo "sleep for 10 seconds"
+  sleep 10
+done
+
+echo "set meta.lb.only_move_primary true"
+echo "remote_command -l $pmeta meta.lb.only_move_primary true" | ./run.sh shell --cluster $meta_list &>/tmp/$UID.$PID.pegasus.online_node.only_move_primary
+set_ok=`grep OK /tmp/$UID.$PID.pegasus.online_node.only_move_primary | wc -l`
+if [ $set_ok -ne 1 ]; then
+  echo "ERROR: meta.lb.only_move_primary true"
+  exit 1
+fi
+echo
+
+echo "Set meta level to lively..."
+echo "set_meta_level lively" | ./run.sh shell --cluster $meta_list &>/tmp/$UID.$PID.pegasus.online_node.set_meta_level
+set_ok=`grep 'control meta level ok' /tmp/$UID.$PID.pegasus.online_node.set_meta_level | wc -l`
+if [ $set_ok -ne 1 ]; then
+  echo "ERROR: set meta level to steady failed"
+  exit 1
+fi
+
+echo "Wait cluster to become balanced..."
+echo "Wait for 3 minutes to do load balance..."
+  sleep 180
+  while true
+  do
+    op_count=`echo "cluster_info" | ./run.sh shell --cluster $meta_list | grep balance_operation_count | grep -o 'total=[0-9][0-9]*' | cut -d= -f2`
+    if [ -z "op_count" ]; then
+      break
+    fi
+    if [ $op_count -eq 0 ]; then
+      echo "Cluster may be balanced, try wait 30 seconds..."
+      sleep 30
+      op_count=`echo "cluster_info" | ./run.sh shell --cluster $meta_list | grep balance_operation_count | grep -o 'total=[0-9][0-9]*' | cut -d= -f2`
+      if [ $op_count -eq 0 ]; then
+        echo "Cluster becomes balanced."
+        break
+      fi
+    else
+      echo "Still $op_count balance operations to do..."
+      sleep 1
+    fi
+  done
+echo
+
+
+
+echo "Set meta level to steady..."
+echo "set_meta_level steady" | ./run.sh shell --cluster $meta_list &>/tmp/$UID.$PID.pegasus.online_node.set_meta_level
+set_ok=`grep 'control meta level ok' /tmp/$UID.$PID.pegasus.online_node.set_meta_level | wc -l`
+if [ $set_ok -ne 1 ]; then
+  echo "ERROR: set meta level to steady failed"
+  exit 1
+fi
+
+echo "set meta.lb.only_move_primary false"
```

Comment:
So the procedure is:

1. meta set only_move_primary = true.
2. rebalance nodes
3. meta set only_move_primary = false.
4. rebalance nodes

I'm confused about why we need step 1 and step 2? As far as I can see, step1+2 rebalance the primaries among the old nodes, but they touch nothing on the new node.